Sources/FronteggSwift/FronteggAuth.swift

```diff
@@ -240,21 +364,33 @@ public class FronteggAuth: ObservableObject {
     }
     
     public func refreshTokenIfNeeded() async -> Bool {
-        guard let refreshToken = self.refreshToken, let accessToken = self.accessToken else {
+        guard let refreshToken = self.refreshToken else {
+            self.logger.info("no refresh token found")
             return false
         }
         
+        self.logger.info("refreshing token")
+        let accessToken = self.accessToken ?? ""
+        
+        DispatchQueue.main.sync {
+            self.refreshingToken=false
+        }
         if let data = await self.api.refreshToken(accessToken: accessToken, refreshToken: refreshToken) {
             await self.setCredentials(accessToken: data.access_token, refreshToken: data.refresh_token)
+            self.logger.info("token refreshed successfully")
+            DispatchQueue.main.sync {
+                self.refreshingToken=false
```

Comment:
self.refreshingToken=false should be `true` I suppose. Because we start token refresh

Reply:
this one must be false, I already fixed the previous one to be true before start refreshing